File: src/main/java/com/vspk/bookrest/email/EmailServiceImpl.java
package com.vspk.bookrest.email;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.mail.javamail.JavaMailSender;
import org.springframework.mail.javamail.MimeMessageHelper;

import javax.mail.MessagingException;

@Slf4j
@RequiredArgsConstructor
public class EmailServiceImpl implements EmailService{
    private final JavaMailSender mailSender;

    @Value("${spring.mail.username}")
    private String username;

    @Override
    public void send(String emailTo, String subject, String message) {
        try {
            var mimeMessage = mailSender.createMimeMessage();
            var helper = new MimeMessageHelper(mimeMessage, "utf-8");
            helper.setText(message, true);
            helper.setTo(emailTo);
            helper.setSubject(subject);
            helper.setFrom(username);
            mailSender.send(mimeMessage);
            log.info("email has been successfully sent to " + emailTo);
        } catch (MessagingException e) {
            log.error("Error, failed to send email");
        }

    }
}

File: src/main/java/com/vspk/bookrest/email/EmailServiceImpl.kt
package com.vspk.bookrest.email

import javax.mail.MessagingException
import lombok.extern.slf4j.Slf4j
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Value
import org.springframework.mail.javamail.JavaMailSender
import org.springframework.mail.javamail.MimeMessageHelper

@Slf4j
class EmailServiceImpl(private val mailSender: JavaMailSender) : EmailService {

    companion object {
        private val log = LoggerFactory.getLogger(EmailServiceImpl::class.java)
    }

    @Value("\${spring.mail.username}")
    lateinit var username: String


    override fun send(emailTo: String, subject: String, message: String) {
        try {
            val mimeMessage = mailSender.createMimeMessage()
            val helper = MimeMessageHelper(mimeMessage, "utf-8")
            helper.setText(message, true)
            helper.setTo(emailTo)
            helper.setSubject(subject)
            helper.setFrom(username)
            mailSender.send(mimeMessage)
            log.info("email has been successfully sent to $emailTo")
        } catch (e: MessagingException) {
            log.error("Error, failed to send email")
        }
    }
}